src/Servers/IIS/IIS/src/Core/IISHttpContext.IHttpRequestLifetimeFeature.cs

```diff
@@ -12,6 +11,8 @@ internal partial class IISHttpContext : IHttpRequestLifetimeFeature
     {
         private CancellationTokenSource _abortedCts;
         private CancellationToken? _manuallySetRequestAbortToken;
+        private object _abortLock = new object();
+        protected bool _requestAborted;
```

Comment:
[In Kestrel this is volatile](https://github.com/aspnet/AspNetCore/blob/d957bc38a5eea8705c8777ae1094e62e6efdd93e/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs#L38). Observing an abort late isn't the end of the world, but since you're removing the Volatile.Reads and Interlocked.Exchanges used to access this before, you might as well make this field volatile.